pkg/roachprod/vm/azure/azure.go

```diff
@@ -1478,39 +1498,48 @@ func (p *Provider) createUltraDisk(
 	return disk, err
 }
 
-// getSubscription chooses the first available subscription. The value
-// is memoized in the Provider instance.
-func (p *Provider) getSubscription(
-	ctx context.Context,
-) (sub subscriptions.Subscription, err error) {
-	sub = func() subscriptions.Subscription {
+// getSubscription returns env.AZURE_SUBSCRIPTION_ID if it exists
+// or the first subscription when listing all available via an API call.
+// The value is memoized in the Provider instance.
+func (p *Provider) getSubscription(ctx context.Context) (string, error) {
+	subscriptionId := func() string {
 		p.mu.Lock()
 		defer p.mu.Unlock()
-		return p.mu.subscription
+		return p.mu.subscriptionId
 	}()
 
-	if sub.SubscriptionID != nil {
-		return
+	if subscriptionId != "" {
+		return subscriptionId, nil
 	}
 
-	sc := subscriptions.NewClient()
-	if sc.Authorizer, err = p.getAuthorizer(); err != nil {
-		return
-	}
+	subscriptionId = os.Getenv("AZURE_SUBSCRIPTION_ID")
 
-	page, err := sc.List(ctx)
-	if err == nil {
-		if len(page.Values()) == 0 {
-			err = errors.New("did not find Azure subscription")
-			return sub, err
+	// Fallback to retrieving the first subscription
```

Comment:
Do we know what order the subscriptions come back in from `sc.List(ctx)` below? Can we guarantee that the first subscription will be a subscription where a roachprod cluster should be allowed to run in?

I'm asking these questions because previously there were few subscriptions in Azure and few that people had access to. Now, there are many more subscriptions and people will generally have access to multiple. Also, soon the revenue and product divisions will be running their roachprod clusters in different subscriptions from engineering.

How about an approach like the following instead?

- If the user specifies an exact subscription to use, use that subscription. If that subscription can't be found, error out (to ensure the cluster doesn't get created in the wrong subscription).
- If the user doesn't set a subscription, try to use a standard set of adhoc subscriptions and use the first one that is found to be accessible to the user. If non exist / are accessible for the user, then it errors. The initial list of subscriptions would be these in this order: `e2e-adhoc`, `Microsoft Azure Sponsorship`.